cmd/auth-rpc-client.go

```diff
@@ -146,20 +146,26 @@ func (authClient *AuthRPCClient) Call(serviceMethod string, args interface {
 	SetToken(token string)
 	SetTimestamp(tstamp time.Time)
 }, reply interface{}) (err error) {
-	// On successful login, attempt the call.
-	if err = authClient.Login(); err == nil {
+	loginAndCallFn := func() error {
+		// On successful login, proceed to attempt the requested service method.
+		if err = authClient.Login(); err != nil {
+			return err
+		}
 		// Set token and timestamp before the rpc call.
 		args.SetToken(authClient.serverToken)
 		args.SetTimestamp(time.Now().UTC())
 
-		// Call the underlying rpc.
-		err = authClient.rpc.Call(serviceMethod, args, reply)
-
-		// Invalidate token, and mark it for re-login on subsequent reconnect.
+		// Finally make the network call using net/rpc client.
+		return authClient.rpc.Call(serviceMethod, args, reply)
+	}
+	if err = loginAndCallFn(); err != nil {
+		// Invalidate token, and mark it for re-login and reconnect.
 		if err == rpc.ErrShutdown {
```

Comment:
You could have `if err = loginAndCallFn(); err == rpc.ErrShutdown {`. No one more level of `if` is needed.